### ISSUE TITLE
Compiler warnings: unused parameter

### ISSUE THREAD

**Issue by richelbilderbeek** [urho3d/Urho3D#723]
When I compile with a high warning level (using GCC 4.9.2) I get many `unused parameter` compiler warnings. This is a bad thing (see references [1,2]). I suggest to either fix this, or let me fix this.

One (of many) example is in `Urho3D/include/Urho3D/Graphics/Direct3D9/D3D9Texture.h`:

```
void SetShadowCompare(bool enable) {}
```

I suggest to replace such a line to:

```
void SetShadowCompare(bool /* enable */) {}
```

Feel free to assign me this task.

## References

[1] Herb Sutter, Andrei Alexandrescu. C++ coding standards: 101 rules, guidelines, and best practices. 2005. ISBN: 0-32-111358-6. Item 1: 'Compile cleanly at high warning levels'.
[2] Linus Torvalds. Re:[PATCH] Don't compare unsigned variable for <0 in sys_prctl(). 2006-11-28. Retrieved on 2010-09-20. 'Friends don't let friends use [gcc] "-W"'

**Comment by cadaver** [comment]
You are welcome to work on this. However, to make sure parameters can be listed correctly by Doxygen (even if it's a no-op function due to graphics API differences) in cases where the empty implementation is in a header file, the parameter name shouldn't be commented out from the header, but rather the function implementation should be moved to a .cpp file, and the parameter name commented out there.

**Comment by richelbilderbeek** [comment]
I agree on the reasoning.

So I will suggest to replace:

```
//in Urho3D/include/Urho3D/Graphics/Direct3D9/D3D9Texture.h
void SetShadowCompare(bool enable) {}
```

To:

```
//in Urho3D/include/Urho3D/Graphics/Direct3D9/D3D9Texture.h
void SetShadowCompare(bool enable);
```

and

```
//in Urho3D/include/Urho3D/Graphics/Direct3D9/D3D9Texture.cpp
void D3D9Texture::SetShadowCompare(bool /* enable */) {}
```

If we agree, feel free to assign me.

**Comment by cadaver** [comment]
The github issue assignment system is a formality we use very little, except for self-assigning tasks. Just make a pull request that references the issue (eg. fixes #723) once the work is done.
